activerecord/lib/active_record/core.rb

```diff
@@ -523,7 +523,7 @@ def inspect
       # We check defined?(@attributes) not to issue warnings if the object is
       # allocated but not initialized.
       inspection = if defined?(@attributes) && @attributes
-        self.class.attribute_names.collect do |name|
+        self.attribute_names.collect do |name|
```

Comment:
Perhaps `self.` can be removed here?

```suggestion
        self.attribute_names.collect do |name|
```